File: src/main/java/com/github/graqr/threshr/model/redsky/products/pdp/client/Enrichment.java
package com.github.graqr.threshr.model.redsky.products.pdp.client;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.github.graqr.threshr.model.redsky.products.plp.search.Images;
import com.github.graqr.threshr.model.redsky.products.plp.search.Video;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;

@Serdeable
public record Enrichment(
        @JsonProperty("buy_url")
        String buyUrl,
        Images images,
        @JsonProperty("nutrition_facts")
        NutritionFacts nutritionFacts,
        List<Video> videos,
        @JsonProperty("return_policies")
        List<ReturnPolicy> returnPolicies) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/product/Enrichment.kt
package com.graqr.threshr.model.redsky.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class Enrichment(
    @param:JsonProperty("buy_url") val buyUrl: String,
    val images: Images,
    @param:JsonProperty("nutrition_facts") val nutritionFacts: NutritionFacts,
    val videos: List<Video>,
    @param:JsonProperty("return_policies") val returnPolicies: List<ReturnPolicy>
) {}
